open-banking-accelerator/components/consent-management/com.wso2.openbanking.accelerator.consent.extensions/src/main/java/com/wso2/openbanking/accelerator/consent/extensions/manage/impl/VRPConsentRequestHandler.java

```diff
@@ -0,0 +1,232 @@
+            }
+
+            if (StringUtils.isEmpty(consentManageData.getHeaders()
+                    .get(ConsentExtensionConstants.X_IDEMPOTENCY_KEY))) {
+                log.error(ErrorConstants.IDEMPOTENCY_KEY_NOT_FOUND);
+                throw new ConsentException(ResponseStatus.BAD_REQUEST, ErrorConstants.IDEMPOTENCY_KEY_NOT_FOUND);
+            }
+
+            //Handle payment initiation flows
+            handlePaymentPost(consentManageData, request);
+
+        } catch (ConsentManagementException e) {
+            log.error("Error occurred while handling the initiation request", e);
+            throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR,
+                    ErrorConstants.PAYMENT_INITIATION_HANDLE_ERROR);
+        }
+    }
+
+    /**
+     * This method is responsible for handling the GET request for retrieving consent initiation details.
+     * It validates the consent ID, checks if the consent exists,verifies if the consent belongs to the
+     * client making the request.
+     *
+     * @param consentManageData Object
+     */
+    @Override
+    public void handleConsentManageGet(ConsentManageData consentManageData) {
+
+        String consentId = consentManageData.getRequestPath().split("/")[1];
+        if (ConsentManageUtil.isConsentIdValid(consentId)) {
+            try {
+                ConsentResource consent = ConsentServiceUtil.getConsentService().getConsent(consentId,
+                        false);
+                // Check whether the client id is matching
+                if (!consent.getClientID().equals(consentManageData.getClientId())) {
+                    // Throws the error if the client Ids mismatch
+                    if (log.isDebugEnabled()) {
+                        log.debug(String.format("ClientIds missmatch. " +
+                                        "Retrieved client id: %s, ConsentmanageData client id: %s",
+                                consent.getClientID(), consentManageData.getClientId()));
+                    }
+
+                    throw new ConsentException(ResponseStatus.BAD_REQUEST,
+                            new String("Invalid client id passed"));
+                }
+
+                JSONObject receiptJSON = (JSONObject) new JSONParser(JSONParser.MODE_PERMISSIVE).
+                        parse(consent.getReceipt());
+                consentManageData.setResponsePayload(ConsentManageUtil
+                        .getInitiationRetrievalResponse(receiptJSON, consent, consentManageData,
+                                ConsentExtensionConstants.VRP));
+                consentManageData.setResponseStatus(ResponseStatus.OK);
+            } catch (ConsentManagementException | ParseException e) {
+                log.error(ErrorConstants.INVALID_CLIENT_ID_MATCH, e);
+                throw new ConsentException(ResponseStatus.INTERNAL_SERVER_ERROR,
+                        ErrorConstants.ACC_INITIATION_RETRIEVAL_ERROR);
```

Comment:
`ACC_INITIATION_RETRIEVAL_ERROR ` shouldn't this be VRP retrieval related error?